catalog-rest-service/src/main/java/org/openmetadata/catalog/secrets/SecretsManagerMigrationService.java

```diff
@@ -0,0 +1,307 @@
+          service.getConnection().getConfig(),
+          service.getServiceType().value(),
+          service.getName(),
+          repository.getServiceType(),
+          true);
+      // avoid reaching secrets manager quotas
+      Thread.sleep(100);
+      service.getConnection().setConfig(null);
+      repository.dao.update(service);
+    } catch (IOException | InterruptedException e) {
+      throw new SecretsManagerMigrationException(e.getMessage(), e.getCause());
+    }
+  }
+
+  private List<ServiceEntityInterface> retrieveNotStoredServices() {
+    return connectionTypeRepositoriesMap.values().stream()
+        .map(this::retrieveServices)
+        .flatMap(List<ServiceEntityInterface>::stream)
+        .collect(Collectors.toList());
+  }
+
+  private List<ServiceEntityInterface> retrieveServices(ServiceEntityRepository<?, ?> serviceEntityRepository) {
+    try {
+      return serviceEntityRepository
+          .listAfter(
+              null,
+              EntityUtil.Fields.EMPTY_FIELDS,
+              new ListFilter(),
+              serviceEntityRepository.dao.listCount(new ListFilter()),
+              null)
+          .getData().stream()
+          .map(ServiceEntityInterface.class::cast)
+          .filter(
+              service ->
+                  !Objects.isNull(service.getConnection()) && !Objects.isNull(service.getConnection().getConfig()))
+          .collect(Collectors.toList());
+    } catch (IOException e) {
+      throw new SecretsManagerMigrationException(e.getMessage(), e.getCause());
+    }
+  }
+
+  private void migrateIngestionPipelines(IngestionPipeline ingestionPipeline) {
+    try {
+      IngestionPipeline ingestion = ingestionPipelineRepository.dao.findEntityById(ingestionPipeline.getId());
+      ingestion.getOpenMetadataServerConnection().setSecurityConfig(null);
+      ingestionPipelineRepository.dao.update(ingestion);
+    } catch (IOException e) {
+      throw new SecretsManagerMigrationException(e.getMessage(), e.getCause());
+    }
+  }
+
+  private List<IngestionPipeline> retrieveNotStoredIngestionPipelines() {
+    try {
+      return ingestionPipelineRepository
+          .listAfter(
+              null,
+              EntityUtil.Fields.EMPTY_FIELDS,
+              new ListFilter(),
+              ingestionPipelineRepository.dao.listCount(new ListFilter()),
+              null)
+          .getData().stream()
+          .filter(
+              ingestionPipeline ->
+                  !Objects.isNull(ingestionPipeline.getOpenMetadataServerConnection())
+                      && !Objects.isNull(ingestionPipeline.getOpenMetadataServerConnection().getSecurityConfig()))
+          .collect(Collectors.toList());
+    } catch (IOException e) {
+      throw new SecretsManagerMigrationException(e.getMessage(), e.getCause());
+    }
+  }
+
+  /** This method delete all the change events which could contain connection config parameters for services */
+  private void deleteChangeEventsForServices() {
```

Comment:
@nahuelverdugo just an FYI, lets make sure we do not ever create a change event if a password or connection is changed in any service

Reply:
We need to create an event change for updating an object, but we are not passing the connection details. https://github.com/open-metadata/OpenMetadata/blob/a6db2e8a84ed1e05f1dbd1b6a756f03a76eb4871/catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/ServiceRepository.java#L157